nixos/modules/services/mail/dovecot.nix

```
@@ -288,6 +395,26 @@ in
   options.services.dovecot2 = {
     enable = mkEnableOption "the dovecot 2.x POP3/IMAP server";
 
+    package = mkPackageOption pkgs "dovecot" {
+      default = [
+        (if versionAtLeast config.system.stateVersion "25.11" then pkgs.dovecot else pkgs.dovecot_2_3)
```

This isn't how you use `mkPackageOption`. This is probably what's causing the CI failure:

> Cacheable portion of option doc build failed.
> Usually this means that an option attribute that ends up in documentation (eg `default` or `description`) depends on the restricted module arguments `config` or `pkgs`.

```suggestion
        (if versionAtLeast config.system.stateVersion "25.11" then "dovecot" else "dovecot_2_3")
```

---

TBH, even reading `config.system.stateVersion` might be too dynamic for the docs. If so, you may need to manually write the `defaultText` for this option:

```nix
lib.mkOption {
  type = lib.types.package;
  description = "The dovecot package to use.";
  default = if versionAtLeast config.system.stateVersion "25.11" then pkgs.dovecot else pkgs.dovecot_2_3;
  defaultText = lib.literalExpression "pkgs.dovecot";
}
```

---

thanks for the review!

fwiw seems like CI didn't like `mkPackageOption` with `defaultText` either, I can still try `pkgsText` later if it doesn't work out like this but it seems to be building fine for now

---

> fwiw seems like CI didn't like `mkPackageOption` with `defaultText` either,

Correct, `mkPackageOption` doesn't take a `defaultText`; `mkPackageOption` exists to write a `defaultText` _for_ you, based on the supplied pkgs attr-path. See https://noogle.dev/f/lib/mkPackageOption

The suggestion to use `defaultText` was if manually calling `mkOption`.